azurerm/helpers/azure/app_service_schedule_backup.go

```diff
@@ -0,0 +1,207 @@
+package azure
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+
+	"github.com/Azure/go-autorest/autorest/date"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2018-02-01/web"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+)
+
+func SchemaAppServiceBackup() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+
+				"storage_account_url": {
+					Type:         schema.TypeString,
+					Required:     true,
+					Sensitive:    true,
+					ValidateFunc: validate.URLIsHTTPS,
+				},
+
+				"enabled": {
+					Type:     schema.TypeBool,
+					Optional: true,
+					Default:  true,
+				},
+
+				"schedule": {
+					Type:     schema.TypeList,
+					Required: true,
+					MaxItems: 1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"frequency_interval": {
+								Type:         schema.TypeInt,
+								Required:     true,
+								ValidateFunc: validateFrequencyInterval,
+							},
+
+							"frequency_unit": {
+								Type:     schema.TypeString,
+								Required: true,
+								ValidateFunc: validation.StringInSlice([]string{
+									"Day",
+									"Hour",
+								}, false),
+							},
+
+							"keep_at_least_one_backup": {
+								Type:     schema.TypeBool,
+								Optional: true,
+								Default:  false,
+							},
+
+							"retention_period_in_days": {
+								Type:         schema.TypeInt,
+								Optional:     true,
+								Default:      30,
+								ValidateFunc: validateRetentionPeriod,
```

Comment:
Would it make more sense to use the build in validation here:
```suggestion
								ValidateFunc: validation.IntBetween(0, 1000),
```